src/ranch.erl

```diff
@@ -155,10 +155,18 @@ start_error(_, Error) -> Error.
 
 -spec stop_listener(ref()) -> ok | {error, not_found}.
 stop_listener(Ref) ->
+	Transport = get_transport(Ref),
+	TransOpts = get_transport_options(Ref),
 	case supervisor:terminate_child(ranch_sup, {ranch_listener_sup, Ref}) of
 		ok ->
 			_ = supervisor:delete_child(ranch_sup, {ranch_listener_sup, Ref}),
-			ranch_server:cleanup_listener_opts(Ref);
+			ranch_server:cleanup_listener_opts(Ref),
+			case erlang:function_exported(Transport, cleanup, 1) of
```

Comment:
No, make it mandatory. No need to check if the function exists. It MUST exist.

Reply:
Ok, I was thinking of compatibility for existing transports. On the other hand, IIRC we introduce a few already with 2.0, so some more are no real problem.